examples/bri-3/src/bri/workgroup/worksteps/api/worksteps.controller.spec.ts

```diff
@@ -0,0 +1,193 @@
+    });
+
+    it('should return new uuid from the created workstep when all params provided', async () => {
+      // Arrange
+      const requestDto = { name: 'name', version: 'version', status: 'status', workgroupId: 'wgid', securityPolicy: 'secPolicy', privacyPolicy: 'privPolicy' } as CreateWorkstepDto;
+
+      // Act
+      const response = await wController.createWorkstep(requestDto);
+
+      // Assert
+      expect(response.length).toEqual(36);
+    });
+  });
+
+  describe('updateWorkstep', () => {
+    it('should throw NotFound if non existent id passed', () => {
+      // Arrange
+      const nonExistentId = '123';
+      const requestDto = { name: 'name', version: 'version', status: 'status', workgroupId: 'wgid', securityPolicy: 'secPolicy', privacyPolicy: 'privPolicy' } as UpdateWorkstepDto;
+
+      // Act and assert
+      expect(async () => {
+        await wController.updateWorkstep(nonExistentId, requestDto);
+      }).rejects.toThrow(new NotFoundException(NOT_FOUND_ERR_MESSAGE));
+    });
+
+    it('should perform the update if existing id passed', async () => {
+      // Arrange
+      const createRequestDto = { name: 'name1', version: 'version1', status: 'status1', workgroupId: 'wgid1', securityPolicy: 'secPolicy1', privacyPolicy: 'privPolicy1' } as CreateWorkstepDto;
+      const newWorkstepId = await wController.createWorkstep(createRequestDto);
+      const updateRequestDto = { name: 'name2', version: 'version2', status: 'status2', workgroupId: 'wgid2', securityPolicy: 'secPolicy2', privacyPolicy: 'privPolicy2' } as UpdateWorkstepDto;
+
+      // Act
+      await wController.updateWorkstep(newWorkstepId, updateRequestDto);
+      
+      // Assert
+      const updatedWorkstep = await wController.getWorkstepById(newWorkstepId);
+      expect(updatedWorkstep.id).toEqual(newWorkstepId);
+      expect(updatedWorkstep.name).toEqual(updateRequestDto.name);
+      expect(updatedWorkstep.version).toEqual(updateRequestDto.version);
+      expect(updatedWorkstep.status).toEqual(updateRequestDto.status);
+      expect(updatedWorkstep.workgroupId).toEqual(updateRequestDto.workgroupId);
+      expect(updatedWorkstep.securityPolicy).toEqual(updateRequestDto.securityPolicy);
+      expect(updatedWorkstep.privacyPolicy).toEqual(updateRequestDto.privacyPolicy);
+    });
+  });
+
+  describe('deleteWorkstep', () => {
+    it('should throw NotFound if non existent id passed', () => {
+      // Arrange
+      const nonExistentId = '123';
+      // Act and assert
+      expect(async () => {
+        await wController.deleteWorkstep(nonExistentId);
+      }).rejects.toThrow(new NotFoundException(NOT_FOUND_ERR_MESSAGE));
+    });
+
+    it('should perform the delete if existing id passed', async () => {
+      // Arrange
+      const createRequestDto = { name: 'name1', version: 'version1', status: 'status1', workgroupId: 'wgid1', securityPolicy: 'secPolicy1', privacyPolicy: 'privPolicy1'} as CreateWorkstepDto;
```

Comment:
optional if you want to do since it is small refactoring, but you can create method
const createTestWorkstep = async () : workstepId => { ... }
where you would put create workstep since you have it in multiple tests

Reply:
creates createWorkstep function and requestDTO object at top of file for reuse in multiple tests